---
output:
  word_document:
        reference_docx: style_reference.docx
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, message = FALSE, warning = FALSE, error = FALSE, cache = TRUE, fig.width = 6.5, dpi = 500)
```

```{r load_libraries_data_functions}
library(DEXSeq)
library(DESeq2)
library(goseq)
library(limma)
library(genefilter)
library(e1071)

library(GenomicFeatures)
library(TxDb.Mmusculus.UCSC.mm10.knownGene)
library(org.Mm.eg.db)
library(GO.db)
library(KEGG.db)
library(pathview)

library(tidyverse)
library(reshape2)
library(cowplot)
library(pheatmap)
library(RColorBrewer)
library(DiagrammeR)
library(stringr)
library(knitr)

load('../processed_data.RData')

source('../R/00.functions.R')
source('../scripts/load_SubreadOutput.R')
```

# SUPPLEMENTARY DATA  

```{r figure_s1, fig.height=3, fig.width=6.5, fig.align='center'}
a <- assay(dds) %>%
  melt %>%
  setNames(c('gene_id', 'sample', 'count')) %>%
  left_join(sample_table) %>%
  filter(gene_id %in% c('Lpl', 'Cebpa', 'Pparg'),
         stage %in% c('Proliferation', 'Differentiation')) %>%
  group_by(gene_id, stage) %>%
  summarise(ave = log(mean(count))) %>%
  ggplot(aes(x = gene_id, y = ave, fill = stage)) +
  geom_col(position = 'dodge', width = .6) +
  theme_classic() +
  labs(x = '', y = 'Average Log Count') +
  theme(legend.position = c(.5, 1), legend.direction = 'vertical', legend.title = element_blank(),
        axis.title.y = element_text(size = 7), axis.text.x = element_text(angle = 45, size = 7, hjust = 1),
        plot.margin = unit(c(1,.5,.8,.2), 'cm'))


ind <- c('Fasn', 'Acly', 'Acaca', 'Elovl6', 'Scd1', 'Scd2', 'Scd3', 'Scd4', 'Dgat1', 'Dgat2')
lab <- c('Scd4', 'Elovl6', 'Scd3', 'Acaca', 'Dgat1', 'Acly', 'Dgat2', 'Fasn', 'Scd2', 'Scd1')
b <- assay(dds) %>%
  melt %>%
  setNames(c('gene_id', 'sample', 'count')) %>%
  left_join(sample_table) %>%
  filter(gene_id %in% ind) %>%
  group_by(gene_id, stage) %>%
  summarise(ave = log(mean(count))) %>%
  ggplot(aes(x = stage, y = ave, group = gene_id, color = gene_id)) +
  geom_line() +
  theme_classic() +
  labs(x = '', y = 'Average Log Count') +
  theme(legend.position = 'none',
        legend.text = element_text(size = 7),
        axis.title.y = element_text(size = 7),
        axis.text.x = element_text(angle = 45, hjust = 1, size = 7),
        plot.margin = unit(c(1,.5,0,.2), 'cm')) +
  annotate('text', x = 4.1, y = c(7.2, 8.2, 9.8, 10.8, 11.2, 11.8, 12.4, 13.4, 14, 15.8), label = lab, size = 2, hjust = 0)

plot_grid(a, b,
          labels = 'AUTO',
          label_size = 10)

```

Figure S1. Characterization of adipocyte differentiation markers. Eight RNA-seq samples of MDI-induced 3T3-L1 pre-adipocytes at four different time points were obtained from the European Nucleotide Archive (ENA). Raw reads were aligned to the USCS mouse reference genome (mm10) using HISAT2, and aligned reads were summarized at the gene level using HTSeq. Average log counts of three differentiation markers (A) in two developmental stages (control proliferating cells and differentiating cells) and  some lipogenesis markers (B) in four developmental stages (proliferation, quiescence, clonal expansion, and differentiation) are shown as bars and lines, respectively. *Cebpa*, CCAAT/enhancer binding protein (C/EBP), alpha; *Lpl*, lipoprotein lipase; *Pparg*, peroxisome proliferator activated receptor gamma; *Acaca*, Acetyl-CoA Carboxylase Alpha; *Acly*, ATP Citrate Lyase; *Dgat*, Diacylglycerol O-Acyltransferase; *Elov6*, Fatty Acid Elongase 6; *Fasn*, Fatty Acid Synthase; *Scd*, Stearoyl-CoA Desaturase.

```{r figure_s2, fig.height=3}
## A) Dispersion
par(mfrow=c(1,3), mar = c(5.1, 4.1, 4.1, 1))

plotDispEsts(dds)
mtext('A', line = 0, adj = -.4)

## B) MDS
sample_dist <- dist(t(assay(dds)))
mds <- data.frame(cmdscale(as.matrix(sample_dist)))
plot(mds$X1, mds$X2, pch = 19, col = sample_table$stage,
     xlab = 'M1', ylab = 'M2')
legend('bottomright', legend = unique(sample_table$stage), pch = 19, col = unique(sample_table$stage), cex = .8)
mtext('B', line = 0, adj = -.4)

## C) validataion
vdf <- res_groups %>%
  filter(stage == 'stageDifferentiation') %>%
  mutate(data_set = 'Al Adhami et al') %>%
  bind_rows(res2) %>%
  filter(row %in% unlist(go[['GO:0006914']])) %>%
  dcast(row ~ data_set, value.var = 'baseMean')

plot(vdf$`Al Adhami et al`, vdf$`Duteil et al`,
     pch = 19,
     xlab = 'Al Adhami et al', ylab = 'Duteil et al')

mtext('C', line = 0, adj = -.4)
```

Figure S2. Quality assessment, exploration and validation of the data. (A) The estimated genes (black), fitted (red) and final dispersion (blue) data of Al Adhami et al [4] calculated by DESeq2 were plotted against the mean normalized counts for numbers of genes across 8 samples. (B) Multidimensional Scaling (MDS) was applied to the count matrix of 8 samples and the two dimensions were mapped to the 4 stages of adipocyte differentiation; proliferation (black), quiescence (red), clonal expansion (green) and differentiation (blue). (C) Six samples of MDI-induced 3T3-L1 at day 0 and day 7 of the other dataset (Duteil et al [6]) were processed using the same pipeline and tested for differential expression and used for validating the main findings. Log fold-changes of 35 autophagy-related genes from the corresponding comparisons (Al Adhami et al and Duteil et al datasets) are shown on the x and y-axis, respectively.  

```{r figure_s3, fig.height=6}
pathways <- c('04150', '04630', '04910', '04920')
names(pathways) <- LETTERS[1:4]
par(mfrow=c(2,2), mar = c(4, 5, 2, 1))
for(i in seq_along(pathways)) {
  p <- get_pathway(res_groups, 'stageDifferentiation', pathways[i], kegg)
  barcodeplot(p$pvalues,
              index = p$index,
              gene.weights = p$weights,
              quantiles = c(.5, .1),
              labels = c("Low", "High"),
              weights.label = 't-stat',
              xlab = 'p-values')
  mtext(names(pathways)[i], line = 0, adj = -.2)
}
```

Figure S3. Enrichment of signaling pathways at differentiated adipocytes. Bar-codes plots of significant enrichment of signaling pathways; mTOR (A) Jak-STAT (B) Insulin (C) and adipocytokine signaling pathway (D). *p*-values are shown on the x-axis ordered from high to low and divided by the cutoffs 0.01 and 0.05, respectively. *p*-values and t-statistics are calculated for the comparison between differentiating (day 6 after treatment) and proliferating cells (day 0). The vertical bars represent the individual genes with their corresponding *t*-statistics; red line (positive) and blue line (negative). The horizontal lines are the overall enrichment of the pathway.  

```{r figure_s4}
plotDEXSeq(res_exon,
           'chr2_Acbd5+',
           fitExpToVar = 'stage',
           expression = FALSE,
           splicing = TRUE,
           displayTranscripts = TRUE,   
           cex = .7, cex.lab = .7, cex.axis = .7, cex.main = .7)
mtext('Proliferation', line = 26, adj = 1, cex = .7, col = 'green')
mtext('Quiescence', line = 25, adj = 1, cex = .7, col = 'purple')
mtext('Clonal Expansion', line = 24, adj = 1, cex = .7, col = 'red')
mtext('Differentiation', line = 23, adj = 1, cex = .7,col = 'blue')
```

Figure S4. Differential exon usage of *Acbd5*. Counts of total 16073 exons in 8 samples were used to test for differential usage using DEXSeq for the for the pairwise comparison between proliferating samples (day 0) and quiescence (day 2), clonal expansion (10 hours after MDI treatment) and differentiating samples (day 6 after MDI treatment). Exon usage values of 19 exons of *Acbd5* at different differentiation stages (*green*, proliferation; *purple*, quiescence; *red*, clonal expansion; *blue*, differentiation) are shown with their corresponding transcripts and genomic ranges at the bottom.  